### PR TITLE
fail PR

### PR DESCRIPTION
**Pull request by xFLy3r**

### CODE REVIEW

File: src/AppBundle/Controller/TicketsController.php [hunk 1]
```diff
@@ -30,6 +31,11 @@ public function getAction(Ticket $ticket)
     public function freeAction(Ticket $ticket)
     {
         $em = $this->getDoctrine()->getManager();
+
+        if ($ticket->getStatus() === Ticket::STATUS_FREE) {
+            throw new TicketStatusConflictException('Ticket is already free');
```
Comment (spolischook) [inline src/AppBundle/Controller/TicketsController.php]: really? Lets do it method idempotent. Remove this check and exception

Comment (xFLy3r) [inline src/AppBundle/Controller/TicketsController.php]: @spolischook , remove check and exception like this from method ticket/reserve too or not?

Comment (spolischook) [inline src/AppBundle/Controller/TicketsController.php]: no

File: src/AppBundle/Controller/TicketsController.php [hunk 2]
```diff
@@ -41,6 +47,11 @@ public function freeAction(Ticket $ticket)
     public function reserveAction(Ticket $ticket)
     {
         $em = $this->getDoctrine()->getManager();
+
+        if ($ticket->getStatus() === Ticket::STATUS_BOOKED) {
```
Comment (spolischook) [inline src/AppBundle/Controller/TicketsController.php]: the test should appear incorrect behavior

File: src/AppBundle/Repository/CustomerOrderRepository.php [hunk 3]
```diff
@@ -2,7 +2,29 @@
...
 class CustomerOrderRepository extends AbstractRepository
 {
-
+    /**
+     * Method returns opened customer order
+     *
+     * @param Customer $param
+     * @return CustomerOrder
+     */
+    public function findOpenedCustomerOrder(Customer $param)
```
Comment (spolischook) [inline src/AppBundle/Repository/CustomerOrderRepository.php]:
```php
public function findLastOpenOrder(Customer $customer)
```

File: src/AppBundle/Repository/CustomerOrderRepository.php [hunk 4]
```diff
@@ -2,7 +2,29 @@
...
+    {
+        return $this->createQueryBuilder('o')
+            ->join('o.customer', 'c')
+            ->where('o.status = :opened')
+            ->andWhere('c.id = :param')
+            ->setParameter(':param', $param->getId())
+            ->setParameter(':opened', 'opened')
```
Comment (spolischook) [inline src/AppBundle/Repository/CustomerOrderRepository.php]: Use constant here

**Pull request closed without merging**